tester/cs/testing-with-gitlab-ci.texy

```diff
@@ -0,0 +1,124 @@
+Průběžné testování s GitLsb CI
+******************************
+
+.[perex]
+Znáte to. Napsali jste testy, které teď procházejí, poplácali se po zádech a váš spánek začal být mnohem klidnější. Máte jistotu, že váš kód spolehlivě funguje. A pak se to stane. Hodiny ukazují 3:15 a do nočního ticha se zakousne nepříjemný zvuk drnčení mobilu na nočním stolku. Volá šéf. Zase to nefunguje. Zalije vás studený pot. Vždyť na to máme testy! V tu chvíli se vynoří vzpomínka na páteční odpoledne - spěcháte domů a zapomínáte spustit testy po poslední úpravě.
+
+V dnešní době se hojně používá GitLab, důvodů je více - intuitivní ovládání GUI, ale hlavní také je výhoda je, že GitLab včetně GitLab CI lze provzovat na vlastním serveru a tím je zaručeno soukromí, protože je opensource. Ale také běž na https://gitlab.com/ .
+
+Pojďme si ukáázat, jak jednodušše přidat CI na GL.
+
+Jak funguje GitLab CI
+=====================
+
+GitLab CI((Continuous Integration)) (dále jen CI) je velmi populární služba pro automatizaci [průběžné integrace](https://cs.wikipedia.org/wiki/Průběžná_integrace). Je poskytována zdarma((Pro soukromé repozitáře má omené CI minuty, neýká se instnce CI na vlastním serveru.)) a nabízí automatizované spouštění testů, deploy aplikací atd. po zápisu změny do repozitáře. Je plně integrována do prostředí GitHubu.
+
+CI se konfiguruje souborem `.gitlab-ci.yml` umístěným v kořenovém adresáři projektu. Obsahuje několik bloků, které si podrobněji popíšeme:
+
+Je potřeba nedefinovat stages, aby CI vědělo, jak se mají úlohy po sobě spouštět. Úlohy se spoští paralelně v jednotlivé stagi.
+
+Každý job by měl mít definován nějaký image, jinak používá image defaultní, definovaný na CI runneru. Lze ovšem definovat i globálně pro všechny joby v rootu konfigurace.
+
+Proměnné prostředí
+------------------
+
+V sekci `variables` nastavujeme další proměnnépro pro prostředí pro běh testů. Pro každou odrážku se skripty provedou samostatně a v jedné odrážce lze definovat více proměnných. Dále v textu budeme používat proměnnou `TESTER_PHP_BIN`, abychom mohli testy pouštět také s parametrem `-p php-cgi`.
+
+```yaml
+variables:
+  - TESTER_PHP_BIN="php"
+  - TESTER_PHP_BIN="php-cgi"
+```
```

Comment:
Asi dovysvětlit, jak je možné, že se `TESTER_PHP_BIN` může nastavit 2x a kde se to projeví.

Reply:
nemůže dvakrát podle mě